examples/flutter_gallery/lib/demo/rally/charts/pie_chart.dart

```diff
@@ -0,0 +1,254 @@
+        ),
+      ]).animate(controller),
+      curve: Curves.decelerate);
+    controller.forward();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _AnimatedRallyPieChart(
+      animation: animation,
+      centerLabel: widget.heroLabel,
+      centerAmount: widget.heroAmount,
+      total: widget.wholeAmount,
+      segments: widget.segments,
+    );
+  }
+}
+
+class _AnimatedRallyPieChart extends AnimatedWidget {
+  const _AnimatedRallyPieChart({
+    Key key,
+    this.animation,
+    this.centerLabel,
+    this.centerAmount,
+    this.total,
+    this.segments,
+  }) : super(key: key, listenable: animation);
+
+  final Animation<double> animation;
+  final String centerLabel;
+  final double centerAmount;
+  final double total;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle labelTextStyle = Theme.of(context).textTheme.body1.copyWith(
+        fontSize: 14.0,
+        letterSpacing: 0.5,
+    );
+
+    return DecoratedBox(
+      decoration: _RallyPieChartOutlineDecoration(
+        maxFraction: animation.value,
+        total: total,
+        segments: segments,
+      ),
+      child: SizedBox(
+        height: 300.0,
+        child: Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              Text(
+                centerLabel,
+                style: labelTextStyle,
+              ),
+              Text(
+                usdWithSignFormat.format(centerAmount),
+                style: Theme.of(context).textTheme.headline,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _RallyPieChartOutlineDecoration extends Decoration {
+  const _RallyPieChartOutlineDecoration({this.maxFraction, this.total, this.segments});
+
+  final double maxFraction;
+  final double total;
+  final List<RallyPieChartSegment> segments;
+
+  @override
+  BoxPainter createBoxPainter([VoidCallback onChanged]) {
+    return _RallyPieChartOutlineBoxPainter(
+      maxFraction: maxFraction,
+      wholeAmount: total,
+      segments: segments,
+    );
+  }
+}
+
+class _RallyPieChartOutlineBoxPainter extends BoxPainter {
+  _RallyPieChartOutlineBoxPainter({this.maxFraction, this.wholeAmount, this.segments});
+
+  final double maxFraction;
+  final double wholeAmount;
+  final List<RallyPieChartSegment> segments;
+  static const double wholeRadians = 2.0 * pi;
+  static const double spaceRadians = wholeRadians / 180.0;
+
+  @override
+  void paint(Canvas canvas, Offset offset, ImageConfiguration configuration) {
+    // Create two padded reacts to draw arcs in: one for colored arcs and one for
+    // inner bg arc.
+    const double strokeWidth = 4.0;
+    final double outerRadius = min(
+      configuration.size.width,
+      configuration.size.height,
+    ) / 2;
```

Comment:
It doesn't really matter, but it might look a little tidier to consistently use integer notation for whole numbers.